src/pmem.c

```diff
@@ -602,17 +624,207 @@ memmove_nodrain_normal(void *pmemdest, const void *src, size_t len)
 	return memmove(pmemdest, src, len);
 
 }
+
+/*
+ * memcpy_nodrain_normal -- (internal) memmcpy to pmem without hw drain
+ */
+static void *
+memcpy_nodrain_normal(void *pmemdest, const void *src, size_t len)
+{
+	return memcpy(pmemdest, src, len);
+}
+
+/*
+ * nodrain_movnt -- (internal) memXXX worker function without hw drain, movnt
+ */
+
+static void *
+nodrain_movnt(void *pmemdest, void *src, size_t len)
+{
+	/* way too chatty for LOG level 3 */
+	LOG(15, "pmemdest %p src %p len %zu", pmemdest, src, len);
+
+	__m128i xmm0, xmm1, xmm2, xmm3, xmm4, xmm5, xmm6, xmm7;
+	size_t cnt = 0;
+	uint32_t i = 0;
```

Comment:
1) There's no need to initialize `i` here. Same for `cnt`.
2) `i` is later compared with `cnt` (i.e. line 669), which is of type *size_t*. In theory, `cnt` could be quite large (len / 128), possibly larger than *uint32_t*. So, I would rather use *size_t* for `i`.